File: dotnet/Example/Sort/Service/Pure/HeapsortService.cs
namespace Example.Sort.Service.Pure
{
    public class HeapsortService : Example.Sort.Service.Pure.SortService
    {
        private Example.Sort.Service.HeapsortService otterop;

        private HeapsortService(Example.Sort.Service.HeapsortService wrapped)
        {
            this.otterop = wrapped;
        }
        public HeapsortService()
        {
            this.otterop = new Example.Sort.Service.HeapsortService();
        }

        public string[] Sort(string[] array)
        {
            var _arrayArray = new Otterop.Lang.String[array.Count()];
            for (var i = 0; i < array.Count(); i++)
            {
                _arrayArray[i] = Otterop.Lang.String.wrap(array[i]);
            }
            var _array = Otterop.Lang.Array.Wrap(_arrayArray);
            var retOtterop = otterop.Sort(_array);
            var ret = new string[retOtterop.Size()];
            for (var i = 0; i < retOtterop.Size(); i++)
            {
                var retI = retOtterop.Get(i);
                ret[i] = retI.unwrap();
            }
            return ret;
        }

        public Example.Sort.Service.HeapsortService unwrap()
        {
            return this.otterop;
        }

        public static Example.Sort.Service.Pure.HeapsortService wrap(Example.Sort.Service.HeapsortService wrapped)
        {
            return new Example.Sort.Service.Pure.HeapsortService(wrapped);
        }
    }

}

File: src/main/java/example/sort/service/pure/HeapsortService.java
package example.sort.service.pure;

public class HeapsortService implements example.sort.service.pure.SortService {
    private final example.sort.service.HeapsortService _otterop;

    private HeapsortService(example.sort.service.HeapsortService wrapped) {
        this._otterop = wrapped;
    }

    public HeapsortService() {
        this._otterop = new example.sort.service.HeapsortService();
    }

    public String[] sort(String[] array) {
        var _arrayArray = new otterop.lang.String[array.length];
        for (var i = 0; i < array.length; i++) {
            _arrayArray[i] = otterop.lang.String.wrap(array[i]);
        }
        var _array = otterop.lang.Array.wrap(_arrayArray);
        var retOtterop = _otterop.sort(_array);
        var ret = new String[retOtterop.size()];
        for (var i = 0; i < retOtterop.size(); i++) {
            var retI = retOtterop.get(i);
            ret[i] = retI.unwrap();
        }
        return ret;
    }

    public example.sort.service.HeapsortService unwrap() {
        return this._otterop;
    }

    public static example.sort.service.pure.HeapsortService wrap(example.sort.service.HeapsortService wrapped) {
        return new example.sort.service.pure.HeapsortService(wrapped);
    }
}

